pkg/experimental/pkg_filegroup.bzl

```diff
@@ -578,6 +591,83 @@ pkg_mkdirs = rule(
             The default is `dir`.
             """,
             default = "dir",
+            # TODO(nacl): use "values" here
+        ),
+    },
+)
+
+def _pkg_mklinks_impl(ctx):
+    _validate_attr(ctx.attr.attrs)
+    return [
+        PackageSymlinkInfo(
+            link_map = ctx.attr.links,
+            attrs = ctx.attr.attrs,
+            section = ctx.attr.section,
+        ),
+    ]
+
+pkg_mklinks = rule(
+    doc = """pkg_filegroup-like-rule for symlink management.
+
+    This rule results in the creation of one or more symbolic links a package.
+    The link may point to a location outside of it.
+
+    """,
+    implementation = _pkg_mklinks_impl,
+    attrs = {
+        "links": attr.string_dict(
+            doc = """Link mappings to create.
+
+            The keys of this dict are the link names, the values are the source
```

Comment:
Do they really have to be source files? You might want the results of a target.
Can they be dangling?

Reply:
> Do they really have to be source files? You might want the results of a target.

I think we may be crossing wires on terminology here.  In this case I'm referring to source-destination mappings within the archives themselves.

I'll take a peek 

>  Can they be dangling?

Yes.

I'll work on some documentation updates.

Reply:
Documentation updated in https://github.com/bazelbuild/rules_pkg/pull/175/commits/4f31b82249c842f01540e477750b84745579a057